### PR TITLE
Opentask view Support

### PR DESCRIPTION
**Pull request by mebitek**
I added Opentask support in read-only mode:
current features
  - show/hide task calendars
  - view task on day, week, month and agenda view
  - view task detail with markdown support
  - support Opentask 
  - share task as VTODO
  - add option to show/hide non-visible calendars in main preferences list
  - widget show tasks

### CODE REVIEW

**Comment by jspricke** [comment]
Wow, thanks for working on this! Can you please clean up your commit into logical patches to make it reviewable?
For example: don't change indention of unrelated code, squash multiple changes to the same code into single logical patches, drop unrelated changes..

**Comment by mebitek** [comment]
@jspricke should be ok now.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,233 +18,239 @@
 */
 -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="ws.xsoh.etar"
```
Comment (jspricke) [inline app/src/main/AndroidManifest.xml:21]: please drop unrelated whitespace changes, for example here.

File: app/src/main/java/com/android/calendar/Event.java [hunk 2]
```diff
@@ -32,15 +32,19 @@
...
+import com.android.calendar.persistence.tasks.DmfsOpenTasksContract;
 import com.android.calendar.settings.GeneralPreferences;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Iterator;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.sql.Timestamp;
+import java.util.*;
```
Comment (jspricke) [inline app/src/main/java/com/android/calendar/Event.java:39]: please don't use asterisk imports.

File: app/src/main/java/com/android/calendar/Event.java [hunk 3]
```diff
@@ -1,18 +1,18 @@
 /*
- * Copyright (C) 2007 The Android Open Source Project
```
Comment (jspricke) [inline app/src/main/java/com/android/calendar/Event.java:2]: unrelated whitespace change.

File: app/build.gradle [hunk 4]
```diff
@@ -108,5 +108,9 @@ dependencies {
     def coroutines_version = "1.6.0"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version"
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutines_version"
+    def markwonVersion = "4.6.2"
```
Comment (jspricke) [inline app/build.gradle:111]: this should probably an extra pull request.

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -34,7 +34,6 @@
     <uses-permission android:name="android.permission.REQUEST_IGNORE_BATTERY_OPTIMIZATIONS"/>
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS"/>
     <uses-permission android:name="org.dmfs.permission.READ_TASKS"/>
-    <uses-permission android:name="org.dmfs.permission.WRITE_TASKS"/>
```
Comment (jspricke) [inline app/src/main/AndroidManifest.xml]: changing a line added in the same PR.

File: app/src/main/java/com/android/calendar/persistence/tasks/AstridCloneTasksContract.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package com.android.calendar.persistence.tasks;
+
+import android.net.Uri;
+import android.provider.BaseColumns;
+
+public class AstridCloneTasksContract  implements BaseColumns {
```
Comment (jspricke) [inline app/src/main/java/com/android/calendar/persistence/tasks/AstridCloneTasksContract.java:6]: looks rather non functional, either squash it with the implementation or drop it from the PR.

File: README.md [hunk 7]
```diff
@@ -5,11 +5,13 @@
   - view task on day and week view
   - view task detail with markdown support
   - share task as VEVENT
+  - support Opentask 
 
-* todo list
+* #### todo list
```
Comment (jspricke) [inline README.md:13]: changing lines in the same PR, please squash.

File: app/src/main/java/com/android/calendar/AllInOneActivity.java [hunk 8]
```diff
@@ -105,7 +105,7 @@
 import ws.xsoh.etar.databinding.AllInOneMaterialBinding;
 import ws.xsoh.etar.databinding.DateRangeTitleBinding;
 
-public class AllInOneActivity extends AbstractCalendarActivity implements EventHandler,	OnSharedPreferenceChangeListener, SearchView.OnQueryTextListener, SearchView.OnSuggestionListener, NavigationView.OnNavigationItemSelectedListener {
+public class AllInOneActivity extends AbstractCalendarActivity implements EventHandler, OnSharedPreferenceChangeListener, SearchView.OnQueryTextListener, SearchView.OnSuggestionListener, NavigationView.OnNavigationItemSelectedListener {
```
Comment (jspricke) [inline app/src/main/java/com/android/calendar/AllInOneActivity.java:110]: unrelated whitespace changes.

File: app/src/main/java/com/android/calendar/settings/MainListPreferences.kt [hunk 9]
```diff
@@ -42,6 +40,8 @@ class MainListPreferences : PreferenceFragmentCompat() {
 
     private var currentCalendars = mutableSetOf<Long>()
 
+    private var VIEW_ALL_PREFERENCE = "viewAllPreference"
```
Comment (jspricke) [inline app/src/main/java/com/android/calendar/settings/MainListPreferences.kt:43]: should probably be a new PR.

**Review by jspricke — CHANGES_REQUESTED** [review]
I've added some comments where you could cleanup the patches, there are probably more cases.

**Comment by mebitek** [comment]
@jspricke could you please provide to me the tabbing information so I can use intellinj auto formatting tool

**Comment by jspricke** [comment]
You could start with: https://github.com/Etar-Group/Etar-Calendar/blob/master/.editorconfig
No idea about intellinj, maybe simply don't commit them.

**Comment by mebitek** [comment]
@jspricke I found how to use the editor config - in which way I should commit these  formatting changes?

**Comment by jspricke** [comment]
> in which way I should commit these formatting changes?

I would prefer if you don't commit them.

**Comment by mebitek** [comment]
@jspricke so I do not change the wrong whitespaces? sorry but it's first time involved in opensource project so I need more instructions. thanks

**Comment by jspricke** [comment]
No worries, we are here to help. Whitespaces are different per project and sometimes even within a project so it is best to not touch them and adopt to the style that is used in the file. Also maybe read up on `git rebase -i` on how to clean up your branch.

**Comment by mebitek** [comment]
@jspricke so what I need to do now?

**Comment by jspricke** [comment]
Fix the comments I added (why did you mark them es resolved without fixing them?)

**Comment by mebitek** [comment]
@jspricke I fixed in my code so I asked how to push the formatting changes.

**Comment by jspricke** [comment]
Did you fix the commits in question as well?

**Comment by mebitek** [comment]
I fiex all the whitespaces related changes in all files I have modified. the three request that I live as unresolved are about opening new PR I guess

**Comment by jspricke** [comment]
that does not answer my question.

**Comment by mebitek** [comment]
@jspricke so I do not understand your question sorry

**Comment by jspricke** [comment]
you need to change the commits to exclude the changes in question. Read up on `git rebase -i.
